alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/task/ScheduledTask.java

```diff
@@ -46,36 +46,41 @@
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
     private final TaskScheduler taskScheduler;
-    private final String taskName;
     private ScheduledFuture<?> future;
 
-    public ScheduledTask(final TaskScheduler taskScheduler, final String taskName) {
+    public static String computeTaskName(Class<? extends ScheduledTask> clazz) {
+        String packageName = clazz.getPackageName();
+        String simpleClassName = clazz.getSimpleName();
+
+        return String.format("Task::Class[%s.%s]", packageName, simpleClassName);
+    }
+
+    public ScheduledTask(TaskScheduler taskScheduler) {
         this.taskScheduler = taskScheduler;
-        this.taskName = taskName;
     }
 
-    public String getTaskName() {
-        return taskName;
+    public String computeTaskName() {
+        return ScheduledTask.computeTaskName(getClass());
```

Comment:
Couldn't we keep getTaskName() and in the constructor of Scheduled task do the following:
`taskName = computeTaskName();`

Reply:
We can do this for ScheduledTask, but not for ProviderTask which extends it. This is because ProviderTask would need ProviderProperties passed to its constructor which is currently part of a different ticket.